### ISSUE TITLE
Add validation to the occurrence payloads

### ISSUE THREAD

**Issue by wedamija** [getsentry/sentry#59289]
Action item from https://www.notion.so/sentry/INC-546-Profiling-issue-occurrences-were-no-longer-being-ingested-by-issue-platform-bcb31d75a05c4f6a903c3d6a8224eca5

We payload validation, but we don't cover the whole message, and we haven't versioned our payload. We should look into using https://github.com/getsentry/sentry-kafka-schemas/ to validate our schema, and look into versioning to help enforce backwards compatibility.

**Comment by vartec** [comment]
I believe this would be basically same effort as https://github.com/getsentry/sentry/pull/58743

**Comment by wedamija** [comment]
@vartec Yeah, probably the case. Needs to include versioning, and validating the message type, unless you're already handling that. Closely related though
